nifi-commons/nifi-utils/src/main/java/org/apache/nifi/processor/util/StandardValidators.java

```diff
@@ -980,4 +980,35 @@ public ValidationResult validate(final String subject, final String value, final
             return new ValidationResult.Builder().subject(subject).input(value).explanation(reason).valid(reason == null).build();
         }
     }
+
+    public static final Validator SINGLE_CHAR_VALIDATOR = (subject, input, context) -> {
+        if (input == null) {
+            return new ValidationResult.Builder()
+                    .input(input)
+                    .subject(subject)
+                    .valid(false)
+                    .explanation("Input is null for this property")
+                    .build();
+        }
+
+        if (input.isEmpty()) {
```

Comment:
I think you can remove that if, because when the input is empty it will produce the same message that you generate in case of `input.length() != 1`.